hs/PostResponse.hs

```diff
@@ -0,0 +1,144 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module PostResponse where
+import Data.List
+import           Text.Blaze ((!))
+import qualified Text.Blaze.Html5 as H
+import qualified Text.Blaze.Html5.Attributes as A
+import Happstack.Server
+import MakeElements
+import MasterTemplate
+import Scripts
+import SVGGen
+
+postResponse :: ServerPart Response
+postResponse =
+    ok $ toResponse $
+        masterTemplate "Courseography - Check My POSt!"
+                    [H.meta ! A.name "keywords"
+                            ! A.content "",
+                            postLinks
+                    ]
+                    (do header "post"
+                        checkPost
+                    )
+                    postScripts
+
+checkPost :: H.Html
+checkPost =  do
+    H.html $ do
+        H.head $ do
+            H.title "Check My Post!"
+        H.nav ! A.id "posts" $ H.ul $ do
+            H.li ! A.id "specialist" $ do
+                H.a ! A.href "" $ "Specialist"
+                H.div ! A.id "spec_creds" $ "(0/12.0)"
+            H.li ! A.id "major" $ do
+                H.a ! A.href "" $ "Major"
+                H.div ! A.id "maj_creds" $ "(0/8.0)"
+            H.li ! A.id "minor" $ do
+                H.a ! A.href "" $ "Minor"
+                H.div ! A.id "min_creds" $ "(0/4.0)"
+        H.div ! A.id "div_specialist" $ do
+            H.p $ do
```

Comment:
The `$` and `do` blocks aren't necessary here. If you just have one item that you're applying a function to, (in this case, `H.p`), then you can just apply that function directly.